### PR TITLE
Cleanup GitHub action workflows

### LINKED ISSUE #10: Setup GitHub Action workflow for testing code before merging

### LINKED ISSUE #11: Setup PyPI project and workflow for automatic deployment

### PR DESCRIPTION
**Pull request by RaczeQ**
Closes #10
Closes #11 
Setup tox and added dummy test for CI.

### CODE REVIEW

File: .github/workflows/_tests.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: tests
+on: workflow_call
+
+jobs:
+  run-tests:
+    name: Run tests 🛠️ on multiple systems 🖥️ and Python 🐍 versions
```
Comment (Calychas) [inline .github/workflows/_tests.yml:6]: These emojis I lof <3

File: .github/workflows/_tests.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: pdm-project/setup-pdm@main
+        name: Setup PDM (Python ${{ matrix.python-version }})
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+          enable-pep582: true
+      - name: Install dependencies
+        run: pdm install -d
+      - name: Run tests with tox
+        run: pdm run pytest tests/
```
Comment (Calychas) [inline .github/workflows/_tests.yml:23]: Where is the tox? I don't follow 😅

File: .github/workflows/full-ci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: pdm-project/setup-pdm@main
+        name: Setup PDM
+        with:
+          python-version: 3.8
+          architecture: x64
+          enable-pep582: true
+      - name: Add PDM Publish plugin
+        run: pdm plugin add pdm-publish
```
Comment (Calychas) [inline .github/workflows/full-ci.yml:24]: I think you don't have to install `pdm-publish`. These commands are already included in pdm.

File: .github/workflows/_tests.yml (hunk 2 above)
Comment (Calychas) [reply .github/workflows/_tests.yml:23]: Or maybe, how does it run?

**Review by Calychas — CHANGES_REQUESTED** [review]

**Comment by RaczeQ** [comment]
closes #2

File: .github/workflows/full-ci.yml (hunk 3 above)
Comment (RaczeQ) [reply .github/workflows/full-ci.yml]: True, skipped this step but had to expand step below

File: .github/workflows/_tests.yml (hunk 2 above)
Comment (RaczeQ) [reply .github/workflows/_tests.yml]: Right, missed it during cleaning

File: .github/workflows/_tests.yml (hunk 1 above)
Comment (RaczeQ) [reply .github/workflows/_tests.yml:6]: Copied it from pypi tutorial 😊

**Comment by RaczeQ** [comment]
closes #7

**Review by Calychas — APPROVED** [review]

**Pull request merged**